data-products/src/main/scala/org/sunbird/analytics/archival/AssessmentArchivalJob.scala

```diff
@@ -0,0 +1,103 @@
+package org.sunbird.analytics.archival
+
+import org.apache.spark.sql.functions.{col, concat, lit, to_json, to_timestamp, weekofyear, year}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.ekstep.analytics.framework.conf.AppConf
+import org.ekstep.analytics.framework.util.{JSONUtils, JobLogger}
+import org.ekstep.analytics.framework.{FrameworkContext, JobConfig, Level}
+import org.sunbird.analytics.archival.util.ArchivalRequest
+
+import java.util.concurrent.atomic.AtomicInteger
+
+case class Period(year: Int, weekOfYear: Int)
+case class BatchPartition(batchId: String, period: Period)
+
+object AssessmentArchivalJob extends optional.Application with BaseArchivalJob {
+
+  private val partitionCols = List("batch_id", "year", "week_of_year")
+  private val columnWithOrder = List("course_id", "batch_id", "user_id", "content_id", "attempt_id", "created_on", "grand_total", "last_attempted_on", "total_max_score", "total_score", "updated_on", "question")
+
+  override def getClassName = "org.sunbird.analytics.archival.AssessmentArchivalJob"
+  override def jobName = "AssessmentArchivalJob"
+  override def jobId: String = "assessment-archival"
+  override def getReportPath = "assessment-archival/"
+  override def getReportKey = "assessment"
+  override def dateColumn = "updated_on"
+
+  override def archiveData(requestConfig: Request, requests: Array[ArchivalRequest])(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): List[ArchivalRequest] = {
+
+    val archivalKeyspace = requestConfig.keyspace.getOrElse(AppConf.getConfig("sunbird.courses.keyspace"))
+    val batchId: String = requestConfig.batchId.getOrElse("")
+    val collId: String = requestConfig.collectionId.getOrElse("")
+    val date: String  = requestConfig.date.getOrElse(null)
+
+    var data = loadData(Map("table" -> requestConfig.archivalTable, "keyspace" -> archivalKeyspace, "cluster" -> "LMSCluster"), cassandraUrl, new StructType())
+
+    data = if (batchId.nonEmpty && collId.nonEmpty) {
```

Comment:
Please check out the variable names. 'collId' does not have a clear meaning. Mention it as 'collectionId' so that other developers can understand it.